build.gradle.kts

```
@@ -1,11 +1,50 @@
+import org.gradle.api.tasks.PathSensitivity.NAME_ONLY
+import org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsRootPlugin.Companion.kotlinNodeJsRootExtension
+import org.jetbrains.kotlin.gradle.targets.js.yarn.yarn
+
 plugins {
     kotlin("multiplatform") version "2.1.10"
     kotlin("plugin.serialization") version "2.1.10"
     id("io.kotest.multiplatform") version "6.0.0.M1"
     distribution
 }
 
+val collectCatalogTypings by tasks.registering(Sync::class) {
+    from("../github-actions-typing-catalog/typings")
+    include("**/action-types.yml")
```

In theory, one can also have `action-types.yaml`:
```suggestion
    include("**/action-types.yml")
    include("**/action-types.yaml")
```

---

Only if you change the `TypesProviding`.
It currently requires that in the catalog it is `action-types.yml` and it is probably also a good idea to enforce consistency where you have the control. :-)

---

The bindings server just implements reading of the typings, and (un)fortunately the docs of this action say:
https://github.com/typesafegithub/github-actions-typing/blob/f86d3187950cdcf9ad0a4db0ef9e57d4addcffb4/README.md?plain=1#L5

We can consider enforcing just one extension, but let's think about it later - formally, YAML does allow both... Could we just add `include("**/action-types.yaml")` to reflect the current state, even though the catalog has no `action-types.yaml` files?

---

I'm not sure what you mean.
For the external typings `collectExternalTypings` you can specify a `yml` or `yaml` which are also both supported by the bindings server.

The part with the `include` is only about the catalog typings and there we only have `yml` and only support `yml` in the binding server.
So adding the `yaml` would just be useless.

---

https://github.com/typesafegithub/github-workflows-kt/blob/main/action-binding-generator/src/main/kotlin/io/github/typesafegithub/workflows/actionbindinggenerator/typing/TypesProviding.kt#L37-L39

---

Ok, you're right about enforcing a single extension when we have control - it will let us avoid checking both extensions in some places. I've just implemented this change that effectively resolves this thread: https://github.com/typesafegithub/github-actions-typing-catalog/pull/47

---

Btw, maybe you want to add a cron trigger to that workflow, so that you are informed about changed inputs or outputs daily? 

Currently, if in a month 10 actions changed their API and then one contributor opens a fine PR for a new action, he will get a failed build with various findings that have nothing to do with his changes, which is a bummer for contributors.

So actually you should maybe split away the changes that check all files for YAML extension and all files for API consistency to a separate workflow that is triggered daily and on workflow dispatch, and change that workflow to do both checks only on files touched in a given PR / Push.

---

Yes, I'm planning to improve stuff like you mention, it's just it was easier to quickly add what we have now.